dexter/src/main/java/com/karumi/dexter/listener/threaddecorator/MainThreadSpec.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter.listener.threaddecorator;
+
+/**
+ * A thread specification to execute passed runnable objects in the main thread
+ */
+class MainThreadSpec implements ThreadSpec {
```

Comment:
This thread spec doesn't run anything on the main thread. It's relying on where it is being executed to run it in that very same thread but doesn't force to be the main thread in any way. I'd either rename it or move the logic to run things on the main thread to this class.